File: AndroidClient/app/src/main/java/com/sidescreen/app/WirelessServiceIdentity.kt
package com.sidescreen.app

import java.security.MessageDigest

/** Stable Bonjour instance identity derived from (but not revealing) the pairing token. */
object WirelessServiceIdentity {
    const val SERVICE_TYPE = "_sidescreen._tcp."

    fun nameForToken(token: ByteArray): String {
        require(token.size == 32) { "pairing token must be 32 bytes" }
        val digest = MessageDigest.getInstance("SHA-256").digest(token)
        val suffix = digest.take(8).joinToString(separator = "") { byte ->
            "%02x".format(byte.toInt() and 0xFF)
        }
        return "SideScreen-$suffix"
    }
}

File: MacHost/Sources/WirelessServiceIdentity.swift
import CryptoKit
import Foundation

enum WirelessServiceIdentity {
    static let serviceType = "_sidescreen._tcp"

    static func name(for token: Data) -> String {
        precondition(token.count == 32, "pairing token must be 32 bytes")
        let digest = SHA256.hash(data: token)
        let suffix = digest.prefix(8).map { String(format: "%02x", $0) }.joined()
        return "SideScreen-\(suffix)"
    }
}
